common/lib/xmodule/xmodule/modulestore/django.py

```diff
@@ -243,9 +245,35 @@ class ModuleI18nService(object):
     i18n service.
 
     """
+    def __init__(self, block=None):
+        """
+        Attempt to load an XBlock-specific GNU gettext translator using the XBlock's own domain
+        translation catalog, currently expected to be found at:
+            <xblock_root>/conf/locale/<language>/LC_MESSAGES/<domain>.po|mo
+        If we can't locate the domain translation catalog then we fall-back onto
+        django.utils.translation, which will point to the system's own domain translation catalog
+        This effectively achieves translations by coincidence for an XBlock which does not provide
+        its own dedicated translation catalog along with its implementation.
+        """
+        self.translator = django.utils.translation
+        if block:
+            xblock_resource = block.unmixed_class.__module__
+            xblock_locale_dir = '/conf/locale'
+            xblock_locale_path = resource_filename(xblock_resource, xblock_locale_dir)
+            xblock_domain = 'django'
```

Comment:
@mattdrayer @nedbat are we going to document this in xBlocks ? I meant, xBlocks should follow the pattern for translation files ` translations/ar/LC_MESSAGES/text.po ` instead of  `conf/locale/ar/LC_MESSAGES/django.po`